src/DurableTask.AzureStorage/Partitioning/TablePartitionManager.cs

```
@@ -0,0 +1,572 @@
+            this.tableServiceClient = new TableServiceClient(this.settings.StorageConnectionString);
+            this.partitionTable = tableServiceClient.GetTableClient(this.partitionTableName);
+            this.tableLeaseManager = new TableLeaseManager(this.partitionTable, this.service, this.settings, this.storageAccountName);
+        }
+
+        
+        /// <summary>
+        /// This method create a new instance of the class TableLeaseManager that represents the worker. 
+        /// And then start the loop that the worker keeps operating on the table. 
+        /// </summary>
+        /// <returns></returns>
+        async Task IPartitionManager.StartAsync()
+        {
+            await Task.Factory.StartNew(() => this.PartitionManagerLoop(this.partitionManagerCancellationSource.Token));
+            this.settings.Logger.PartitionManagerInfo(
+                this.storageAccountName,
+                this.settings.TaskHubName,
+                this.settings.WorkerId,
+                "",//empty partition id
+                $"Worker {this.settings.WorkerId} starts working on acquiring and balancing leases.");
+        }
+
+        async Task PartitionManagerLoop(CancellationToken token)
+        {
+            while (!token.IsCancellationRequested)
+            {
+                TimeSpan timeToSleep = this.settings.LeaseRenewInterval;
+                try
+                {
+                    ReadTableReponse response = await this.tableLeaseManager.ReadAndWriteTable();
+                    if (response.WorkOnRelease || response.WaitForPartition)
+                    {
+                        timeToSleep = TimeSpan.FromSeconds(1);
+                    }
+                }
+                catch
+                {
+                    // if the worker failed to update the table, re-read the table immediately without waiting.
+                    timeToSleep = TimeSpan.FromSeconds(0);
+                }
```

Is it safe to just ignore this exception? Shouldn't we consider logging it?

---

This exception is thrown when the worker fails to update the partition table. It is very likely to happen since multiple tables read and write on the same table. So, if a worker failed to update the table, it means that the information it got from the table has already been outdated. It needs to re-read immediately to get the latest information and then decide what it needs to do.